genomesubsampler/prototypeSelection.py

```diff
@@ -114,30 +169,27 @@ def prototype_selection_exhaustive(dm, num_prototypes,
     def prototype_selection_exhaustive(dm: DistanceMatrix, num_prototypes: int,
     max_combinations_to_test: int=200000) -> List[str]:
     '''
-    if num_prototypes < 2:
-        raise ValueError(("'num_prototypes' must be >= 2, since a single "
-                          "prototype is useless."))
-    if num_prototypes >= len(dm.ids):
-        raise ValueError(("'num_prototypes' must be smaller than the number of"
-                          " elements in the distance matrix, otherwise no "
-                          "reduction is necessary."))
-
-    num_combinations = sp.special.binom(len(dm.ids), num_prototypes)
+    _validate_parameters(dm, num_prototypes, seedset)
+
+    ids = set(dm.ids) - set(seedset)
+    num_prototypes = num_prototypes - len(seedset)
+    num_combinations = sp.special.binom(len(ids), num_prototypes)
     if num_combinations >= max_combinations_to_test:
         raise RuntimeError(("Cowardly refuse to test %i combinations. Use a "
                             "heuristic implementation for instances with more "
                             "than %i combinations instead!")
                            % (num_combinations, max_combinations_to_test))
 
+    seedset = tuple(seedset)
     max_dist, max_set = -1 * np.infty, None
-    for s in set(combinations(dm.ids, num_prototypes)):
-        d = distance_sum(s, dm)
+    for s in set(combinations(ids, num_prototypes)):
+        d = distance_sum(s + seedset, dm)
```

Comment:
You are right, but in a different context. Here, we iterate through all exponential many final sets of prototypes and pick the optimal one. Therefore, it won't change the choice on the optimal set if we only compute distance_sum(s), i.e. on a part of the set, because the preselected elements of seedset are guaranteed to be in the result. Thus, it really is only a waste of compute.
This is different from situations where we iteratively construct the prototype set element by element.